### DISCUSSION TITLE
Changing the camera resolution

### DISCUSSION THREAD

**Discussion by dalogue1** [monkeymademe/CamUI#70] [General]
It appears like changing the camera resolution does not take effect until the camera settings are saved and reloaded or the Pi is restarted.

I tested this in the interface and also tried the apply button from Orientation to see if that applied the change but it did not.

I expect that the camera needs to be restarted when resolution is changed.  Perhaps you can restart the camera immediately when the resolution is changed.

**Comment by monkeymademe** [comment]
Yeah this one has been a thorn in my side since at the start of the project where I could not seem to set the image correctly... I sorted it now so thanks for the kick up the butt.

The issue was not that you needed to reload the server it was only the page you needed to reload. Of course the image when its massive is resized to fix the window (responsive layout) but thats only in that window.. In reality the resolution is correct when you take an image. 

If you had a resolution smaller than the interface (say 640x480) then it started to do weird things. If you reloaded the page then its fine. 

Does not matter now its working.

You might have already noticed also in Camera info, (I'll move this to a better named place) you can set the sensor mode that also has a resolution. Don't mistake that for the feed/capture resolution... This is the pixel resolution coming from the camera. Some modes have a scaler crop applied (hence the issues with scaler crop). So you can set a 640x480 sensor mode and get that, but scale it up using the feed/capture resolution setting. You can also do the opposite take the full sensor and scale it down to the lowest feed/capture resolution...

Took me some mental gymnastics to work that out when making this tool!

v3 cameras only have the resolutions that are supported in the sensor mode. So there are 3 resolutions. next task is to make a few smaller custom ones. Its in roadmap

So yeah pull from main it should be working now... sorry for the TMI reply

**Reply by dalogue1** [reply]
> Sorry got for TMI reply

Thanks for the update.  I am interested in the details of solving the issues.  That is part of the curiosity of a hobby. 

It works flawlessly.
